File: 102-binary-tree-level-order-traversal/102-binary-tree-level-order-traversal.cpp
 /**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) 
    {
        //initial 
        vector<vector<int>>ans;
        queue<TreeNode *>q;
        q.push(root);
        if(root==NULL) 
            return ans;
        
        while(1)
        {
            vector<int>data;
            int sz=q.size();
            if(sz==0)
                return ans;
            while(sz>0)
            {
                TreeNode* tmp=q.front();
                q.pop();
                data.push_back(tmp->val);
                
                if(tmp->left!=NULL)
                    q.push(tmp->left);
                if(tmp->right)
                    q.push(tmp->right);   
                sz--;
            }
            ans.push_back(data);
        }
        return ans;
    }
};
       /* vector<vector<int>> ans; //ans
        if(root==NULL) return ans;
        
        queue<TreeNode*> q;
        q.push(root);     
//         
        while(!q.empty())
        {
            int size=q.size();            
            vector<int> currLevel;
            
            while(size-- >0)
            {
                TreeNode *currNode=q.front();
                q.pop();
                if(currNode->left !=NULL) q.push(currNode->left);
                if(currNode->right !=NULL)q.push(currNode->right);
                
                currLevel.push_back(currNode->val);
                    
            }    
            ans.push_back(currLevel);
        }
        return ans;        
    }
};*/

File: 102-binary-tree-level-order-traversal/102-binary-tree-level-order-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>>res=new ArrayList<>();  //ans
        if(root==null) return res;
        
        Queue<TreeNode>q =new LinkedList<>();
        q.add(root);
        
        while(!q.isEmpty()){
            int size=q.size();
            List<Integer>currLevel=new ArrayList<>();
            
            while(size-- >0){
                TreeNode currNode=q.poll(); //front polled from queue ,stored to currNode
                currLevel.add(currNode.val);
                
                if(currNode.left!=null) q.add(currNode.left);
                if(currNode.right!=null) q.add(currNode.right);
                               
            }
            res.add(currLevel);
        }       
        return res;
    }
}